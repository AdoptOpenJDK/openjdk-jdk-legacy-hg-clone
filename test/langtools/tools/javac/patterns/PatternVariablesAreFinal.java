/*
 * @test /nodynamiccopyright/
 * @bug 8231827
 * @summary Ensure that in type test patterns, the predicate is not trivially provable false.
 * @compile/fail/ref=PatternVariablesAreFinal.out -XDrawDiagnostics --enable-preview -source ${jdk.version} PatternVariablesAreFinal.java
 */
public class PatternVariablesAreFinal {
    public static void main(String[] args) {
        Object o = 32;
        if (o instanceof String s) {
            s = "hello again";
            System.out.println(s);
        }
        System.out.println("test complete");
    }
}
