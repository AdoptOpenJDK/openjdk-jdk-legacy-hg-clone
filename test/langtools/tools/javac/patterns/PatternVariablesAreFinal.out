PatternVariablesAreFinal.java:11:13: compiler.err.pattern.binding.may.not.be.assigned: s
- compiler.note.preview.filename: PatternVariablesAreFinal.java
- compiler.note.preview.recompile
1 error
